### DISCUSSION TITLE
Multi Step (wizard) form with rhf and zodschema with only one `useForm` call

### DISCUSSION THREAD

**Discussion by bryanprimus** [react-hook-form/react-hook-form#8926] [Q&A]
All example I found is a `multi-step form` with more than one `useForm` call because they try to maintain or persist the data within different routes and will store the data in some browser storage to make it persist when the user reloads or leaves the browser and gets back.

But In my case, I don't want to persist the data in a global state or browser storage. 

So I try it myself. and ended up with this implementation where I conditionally pass the zod schema to `useForm` for each step.
https://stackblitz.com/github/bryantobing12/vite-rhf-basic-stepper/tree/feat/zod-integration?file=src%2FApp.tsx
![image](https://user-images.githubusercontent.com/46083126/186897825-0df8af0b-e12b-4dd8-baa7-10b066165ad4.png)

I get the idea from `@mantine/form` where they conditionally pass validation based on what step is active. https://mantine.dev/form/recipes/#form-with-multiple-steps

I try to replicate what mantine does with rhf default resolver with conditionally assign errors.
https://stackblitz.com/github/bryantobing12/vite-rhf-basic-stepper?file=src%2FApp.tsx
![image](https://user-images.githubusercontent.com/46083126/186897663-4bc59320-03df-42a0-80ce-2ef7c8e102d6.png)

It seems that everything is working. But I'm still not sure if this is the right implementation or not, or is there a better/correct approach to achieve this.

**Comment by bluebill1049** [comment]
Is it possible to divide your multiple steps into multiple forms (useForm), I find it easier to manage?

**Reply by bryanprimus** [reply]
It's possible but i found it harder. Because i need to merge all the data from every form (useForm) when submitting in the last step. 

I found it harder because i couldn't find the right way to do so (merging the data of each useForm)

From the tutorial you made, you use another state management (little-state-machine) because you want to persist the data to maintain the data when user refresh, 

but in my case i don't need to persist the data to web storage, and i don't want to create another state outside the useForm.

**Reply by bluebill1049** [reply]
You don't need LSM (web stroage), you can replace it with `useState` as long as each form contains its own state and validation.

**Reply by bryanprimus** [reply]
well my intention is to not create another state management (useState,redux, context, lsm) other than the state that `useForm` handle.

may i ask your opinion about the approach that i take? is there a downside of conditioning the schema?  i just find it simpler for my usecase
